### PR TITLE
Add options for COCO reader

### PR DESCRIPTION
**Pull request by awolant**
Signed-off-by: Albert Wolant <awolant@nvidia.com>

### CODE REVIEW

File: dali/pipeline/operators/reader/coco_reader_op.cc [hunk 1]
```diff
@@ -309,6 +309,13 @@ Tensor (`m` * `[x, y, w, h] or `m` * [left, top, right, bottom]`) and labels as
   .AddOptionalArg("ratio",
       R"code(If true, bboxes returned values as expressed as ratio w.r.t. to the image width and height. Default: False)code",
       false)
+  .AddOptionalArg("size_threshold",
+      R"code(If width or height of a bounding box representing an instance of an object is under this value, object will be skipped during reading. It is represented as absolute value. Default value is 0.1)code",
+      0.1f,
+      false)
+  .AddOptionalArg("skip_empty",
+      R"code(If true, reader will skip samples with no object instances in them)code",
```
Comment (Kh4L) [inline dali/pipeline/operators/reader/coco_reader_op.cc]:
```suggestion
      R"code(If true, reader will skip samples with no object instances in them. Default: False)code",
```

Comment (ptrendx) [reply dali/pipeline/operators/reader/coco_reader_op.cc]: That should automatically be populated in the docs, no?

Comment (JanuszL) [reply dali/pipeline/operators/reader/coco_reader_op.cc]: It will be. I think we need to go through operator description and remove this `Default` note

**Comment by awolant** [comment]
Builds 616153

File: dali/pipeline/operators/reader/coco_reader_op.cc (hunk 1 above)
Comment (awolant) [reply dali/pipeline/operators/reader/coco_reader_op.cc]: I removed this unnecessary `Default` info for this op.

**Comment by awolant** [comment]
Builds 616703

File: dali/pipeline/operators/reader/coco_reader_op.cc [hunk 2]
```diff
@@ -428,6 +435,13 @@ void ParseAnnotationFilesHelper(std::vector<std::string> &annotations_filename,
         r.SkipValue();
       }
     }
+    if (skip_empty) {
+      std::vector<std::pair<std::string, int>> non_empty_ids;
```
Comment (mzient) [inline dali/pipeline/operators/reader/coco_reader_op.cc]: You can `reserve` this to accommodate all non-empty entries.

File: dali/pipeline/operators/reader/coco_reader_op.cc [hunk 3]
```diff
@@ -428,6 +435,13 @@ void ParseAnnotationFilesHelper(std::vector<std::string> &annotations_filename,
...
+      for (const auto &id_pair : image_id_pairs)
+        if (annotations_multimap.count(id_pair.second) > 0)
+          non_empty_ids.push_back(id_pair);
+      image_id_pairs = non_empty_ids;
```
Comment (mzient) [inline dali/pipeline/operators/reader/coco_reader_op.cc]:
```suggestion
      image_id_pairs = std::move(non_empty_ids);
```

Comment (awolant) [reply dali/pipeline/operators/reader/coco_reader_op.cc]: Done

File: dali/pipeline/operators/reader/coco_reader_op.cc (hunk 2 above)
Comment (awolant) [reply dali/pipeline/operators/reader/coco_reader_op.cc]: Done

**Comment by awolant** [comment]
Builds 616800

**Pull request merged**